inspector-axon/src/main/java/io/axoniq/inspector/messaging/RollingCountMeasure.kt

```diff
@@ -16,35 +16,33 @@
 
 package io.axoniq.inspector.messaging
 
-import java.util.concurrent.atomic.AtomicLong
+import java.util.concurrent.ConcurrentHashMap
 
 /**
  * Rolls up for every report, which is sent every ten seconds.
  * Since it keeps 6 values, we get the amount of the last minute, updating every 10 seconds
  */
 class RollingCountMeasure {
-    private var count1 = AtomicLong(0)
-    private var count2 = AtomicLong(0)
-    private var count3 = AtomicLong(0)
-    private var count4 = AtomicLong(0)
-    private var count5 = AtomicLong(0)
-    private var count6 = AtomicLong(0)
+    private val countMap = ConcurrentHashMap<Long, Int>()
 
     fun increment() {
-        count1.incrementAndGet()
+        countMap.compute(currentBucket()) { _, v -> (v ?: 0) + 1 }
     }
 
-    fun incrementWindow() {
-        count6 = count5
-        count5 = count4
-        count4 = count3
-        count3 = count2
-        count2 = count1
-        count1.set(0)
+    fun pruneData() {
+        val bucket = currentBucket()
+        val toRemove = countMap.keys.filter { it < (bucket - 60000) }
+        toRemove.forEach { countMap.remove(it) }
     }
 
-    fun value(): Double {
-        val total = count1.get() + count2.get() + count3.get() + count4.get() + count5.get() + count6.get()
-        return total.toDouble()
+    fun count(): Double {
+        val bucket = currentBucket()
+        val minimumBucket = bucket - 60000
+        return countMap.filter { it.key in minimumBucket until bucket }.values.sum().toDouble()
+    }
+
+    private fun currentBucket(): Long {
+        val time = System.currentTimeMillis()
+        return time - time.mod(2500)
```

Comment:
I'm not sure I understand this function. I would expect something like `(time/2500) * 2500` such that within 2,5 seconds it would return the same value.

Reply:
That's not the same, mod will give a value between 0 and 2499, changing every millisecond.

Reply:
Let's say I have a time of 6300. I modulo that, giving 1300. I subtract that of 6300. I get 5000, which is a bucket of 2500

Reply:
Yes, I get it now, really need a vacation. Thanks for explaining.

Reply:
But, I'm fine with the division variant as well. I also just realised, that we don't really need to actual time. Give me a second.